### DISCUSSION TITLE
mrv2 v1.0.7

### DISCUSSION THREAD

**Discussion by ggarra13** [ggarra13/mrv2#196] [Announcements]
mrv2 v1.0.7
=========

- Slow code clean up.
- Faster windows minimal compilations for quickly checking on GitHub.
- Added optional component installation on Windows's .exe installer.  Currently,
  you can install:
  
      * mrv2 application (obviously)
      * mrv2 documentation
      * mrv2 Python demos
      * mrv2 Python libraries
      * mrv2 USD components

   If you choose to install ONLY the application, its size is 395Mb, instead
   of the whole 600Mb.

- Changed name of install directory of mrv2 on Windows from "mrv2 vX.X.X" to "mrv2-vX.X.X" to avoid spaces for easier scripting.
- Fixed image sequence detection that could happen randomly depending on how
  the filesystem returned the order of files.
- Added listing the OS and distro versions at the start and in About of mrv2
  for debugging purposes.
- Made panels be listed alphabetically, regardless of Natural Language.
- Automated version update in docs.
- Removed view sources from html docs to save space on disk.
- Cleaned up docs directory before building docs.
- Sped up creation of docs in build system.
- Updated Python API with FFmpeg and Background changes.
- Fixed reading permissions on files as they were broken!  I did not notice
  as I was using an NTFS drive which sets umask 0022 by default.
- Made reading session more robust to handle missing files or wrong settings.
- Made checking for readable files faster.
- Fixed saving large images/movies with annotations on work properly.
- Improved memory consumption of saving movies, particularly large ones like
  4K.
- Improved logging system logging the tlRender warnings.
- Improved logging system adding a Status mode to report information right
  away.
- FFmpeg's logging now prints out the codec/module where the error was
  generated.
- Improved movie save reporting of parameters.
- Fixed saving a movie with annotations when the movie is bigger than the
  viewport.
- Added GBR8/9/10/12 reading support for VPX.
- Fixed start and end timeline buttons not refreshing thumbnails in the Panels.
- Fixed image panel not refreshing its information when changing images and
  the playback was stopped.
- Fixed Preferences->Positioning->Position/Size when both were used.
- Saving of .otio files as movies when the first clip did not start at 0 now
  works properly.
- Upgraded to RtAudio v5.2.0 on all platforms.
- Added a "View/Auto Frame" to turn off auto framing of the view when changing
  from one clip to another.
- Added User Interface->View->Auto Frame to preferences. 
- Made saving a VP9 or AV1 with the wrong extension not fail.  Instead, they
  are renamed to .mp4.
- Made mrv2's file requester favorites listing automatically remove
  non-existent directories.
- When saving a sequence of OpenEXR, we encode the speed in the actual
  image file, as taken from the playback tool bar.  This value will take
  precedence over the Sequence Default speed as set in Preferences->Playback.
- Made warnings also show up in the status bar, but with an orange background.
- Improved HUD Attributes.  They are now listed alphabetically and they are not
  repeated.  Also, they refresh properly.
- Fixed Media Information panel not refreshing properly when changing images.
- mrv2's tlRender library now reads the video and audio stream metadata.
- When there's no audio metadata there's no longer the titles of
  Attribute/Value used at the end of the Metadata tab in the Media Info Panel.
- Metadata in Media Info Panel is now sorted and stripped of repeated data.
- Media Info Panel's tabs now remember whether they were opened and closed.
- Added GoPro Cineform codec to the list of profiles you can use to encode.
  The biggest benefit is that it can encode at GBRP_12LE and GBRAP_12LE (ie.
  RGB and RGBA at 12 bits with alpha) for a replacement to Apple's proprietary
  ProRes4444 codec.
  It is supported by Adobe Suite (Premiere, After Effects, Media Encoder),
  Apple's Final Cut Pro, Blackmagic Design Davinci Resolve, Avid Media
  Composer, Nuke, Flame and Inferno.

<hr /><em>This discussion was created from the release <a href='https://github.com/ggarra13/mrv2/releases/tag/v1.0.7'>mrv2 v1.0.7</a>.</em>

**Comment by BigRoy** [comment]
> Added optional component installation on Windows's .exe installer.

Installing on Windows without USD can't launch the application.

I get these prompts on launch.
![image](https://github.com/ggarra13/mrv2/assets/2439881/556b5ad6-ee4f-4176-bba7-3fde9391fc8c)
![image](https://github.com/ggarra13/mrv2/assets/2439881/182b8020-6ea5-4a29-9e97-e6c08627a960)
![image](https://github.com/ggarra13/mrv2/assets/2439881/7ea98bd9-85d3-4683-8f63-3b96a2864b40)
![image](https://github.com/ggarra13/mrv2/assets/2439881/9322420d-1697-4c71-88c7-3ee7cb511a78)

---

For completeness: I installed without USD, without documentation and without Python demos in this one.

**Comment by ggarra13** [comment]
Thanks.  I will be re-calling the release.

**Comment by ggarra13** [comment]
Fixed.

**Reply by BigRoy** [reply]
Haha - I suppose disallowing the disabling of USD dependencies was the easiest fix. Anyway, fine with me. Thanks.

**Reply by ggarra13** [reply]
While we can't get rid of USD on install, I *was* able to shave off 120Mb of the mrv2 installer as I was incorrectly shipping a directory of unneded data for USD.

**Reply by BigRoy** [reply]
That's a solid improvement overall, about 20%.

**Comment by ggarra13** [comment]
I discovered two other ugly bugs after release.  One that effects colors when EXRs are saved as movie files getting saved always in 8-bits (which was there since v1.0.5) and a minor one if a movie file was saved with a number in it.

I have upgraded v1.0.7 again.  Please download it again.
